File: src/main/java/org/cobbzilla/util/http/ApiConnectionInfo.java
package org.cobbzilla.util.http;

import static org.cobbzilla.util.reflect.ReflectionUtil.copy;

public class ApiConnectionInfo {

    private String baseUri;
    private String user;
    private String password;

    public ApiConnectionInfo (String baseUri) { this.baseUri = baseUri; }

    public ApiConnectionInfo (ApiConnectionInfo other) { copy(this, other); }

    @java.beans.ConstructorProperties({"baseUri", "user", "password"})
    public ApiConnectionInfo(String baseUri, String user, String password) {
        this.baseUri = baseUri;
        this.user = user;
        this.password = password;
    }

    public ApiConnectionInfo() {
    }

    // alias for when this is used in json with snake_case naming conventions
    public String getBase_uri () { return getBaseUri(); }
    public void setBase_uri (String uri) { setBaseUri(uri); }

    public java.lang.String toString() {
        return "ApiConnectionInfo(baseUri=" + this.baseUri + ", user=" + this.user + ")";
    }

    public String getBaseUri() {
        return this.baseUri;
    }

    public String getUser() {
        return this.user;
    }

    public String getPassword() {
        return this.password;
    }

    public void setBaseUri(String baseUri) {
        this.baseUri = baseUri;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}

File: src/main/java/org/cobbzilla/util/http/ApiConnectionInfo.kt
package org.cobbzilla.util.http

import org.cobbzilla.util.reflect.ReflectionUtil.copy

class ApiConnectionInfo {

    var baseUri: String? = null
    var user: String? = null
    var password: String? = null

    // alias for when this is used in json with snake_case naming conventions
    var base_uri: String?
        get() = baseUri
        set(uri) {
            baseUri = uri
        }

    constructor(baseUri: String) {
        this.baseUri = baseUri
    }

    constructor(other: ApiConnectionInfo) {
        copy(this, other)
    }

    @java.beans.ConstructorProperties("baseUri", "user", "password")
    constructor(baseUri: String, user: String, password: String) {
        this.baseUri = baseUri
        this.user = user
        this.password = password
    }

    constructor() {}

    override fun toString(): java.lang.String {
        return "ApiConnectionInfo(baseUri=" + this.baseUri + ", user=" + this.user + ")"
    }
}
